src/main/java/at/ac/tuwien/kr/alpha/grounder/NaiveGrounder.java

```diff
@@ -96,15 +119,18 @@ public NaiveGrounder(Program program, AtomStore atomStore, boolean debugInternal
 		super(filter, bridges);
 		this.atomStore = atomStore;
 
-		programAnalysis = new ProgramAnalysis(program);
-		analyzeUnjustified = new AnalyzeUnjustified(programAnalysis, atomStore, factsFromProgram);
+		this.applyProgramTransformations(program, useCountingGrid);
+		// TODO it might make more sense to do this outside the grounder, but where
```

Comment:
This looks again like a half-refactored intermediate from #185. If possible, remove/revert it please.